capstone/src/main/java/com/google/sps/servlets/UsersServlet.java

```diff
@@ -0,0 +1,294 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+//import statements for the Users API
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+//import statements for SQL Interaction
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.sql.DataSource;
+
+@WebServlet("/users/*")
+public class UsersServlet extends HttpServlet {
+  private final String CREATE_NEW_USER_QUERY =
+    "INSERT INTO user_info (user_name, full_name, email, verified, university_id) VALUES (?, ?, ?, true, ?)";
+
+  private final String URL_TO_REDIRECT_AFTER_USER_LOGS_OUT = "/";
+  private final String URL_TO_REDIRECT_AFTER_USER_LOGS_IN = "/login";
+
+  private final String GET_USER_ROW_QUERY =
+    "SELECT user_info.id, user_info.email,  user_info.full_name, " +
+    "user_info.user_name, university.name " +
+    "FROM user_info JOIN university ON university.id = user_info.university_id " +
+    "WHERE user_info.email = ?";
+
+  /**
+   * This method is called when a GET request is sent to the "/users/*" endpoint
+   * @param request A HttpServletRequest object that holds the information sent by the user.
+   * @param response A HttpServletResponse object that holds the information that will be sent back to the user.
+   * @return void
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+    throws IOException, ServletException {
+    response.setContentType("application/json");
+    ServletContext servletContext = getServletContext();
+    DataSource pool = (DataSource) servletContext.getAttribute("my-pool");
+    if (!request.getPathInfo().equals("/current")) {
+      response.sendError(404, "Invalid url supplied by the client.");
+      return;
+    }
+    UserService userService = UserServiceFactory.getUserService();
+    Boolean isUserLoggedIn = userService.isUserLoggedIn();
+    Boolean IsUserRegistered;
+    if (!isUserLoggedIn) {
+      response.sendRedirect("/login");
+      return;
+    }
+    try {
+      IsUserRegistered = IsUserRegistered(pool, userService, response, request);
+      if (!IsUserRegistered) {
+        response.setStatus(300);
```

Comment:
I've never seen anyone use a plain 300 error code, usually you just see `301` or `307` for redirects. Probably could be simplified even more but at this point let's just leave this as is.

Reply:
Okay